modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/direct/TransportGetDatabaseConfigurationAction.java

```diff
@@ -82,25 +89,31 @@ protected List<DatabaseConfigurationMetadata> createActionContext(Task task, Get
                 "wildcard only supports a single value, please use comma-separated values or a single wildcard value"
             );
         }
-
+        ProjectId projectId = projectResolver.getProjectId();
         List<DatabaseConfigurationMetadata> results = new ArrayList<>();
-        PersistentTasksCustomMetadata tasksMetadata = PersistentTasksCustomMetadata.getPersistentTasksCustomMetadata(
-            clusterService.state()
+        PersistentTasksCustomMetadata tasksMetadata = PersistentTasksCustomMetadata.get(
+            clusterService.state().metadata().getProject(projectId)
         );
+        String geoIpTaskId = GeoIpDownloaderTaskExecutor.getTaskId(projectId, projectResolver.supportsMultipleProjects());
+        ProjectMetadata projectMetadata = projectResolver.getProjectMetadata(clusterService.state());
```

Comment:
Can you use this `projectMetadata` for retrieving the `PersistentTasksCustomMetadata` and its ID for `GeoIpDownloaderTaskExecutor.getTaskId()`? That way we only need to resolve the project ID once.